gazebo_ros/scripts/spawn_entity.py

```diff
@@ -0,0 +1,354 @@
+        # parser.add_argument('-b', dest='bond', action='store_true', help='bond to gazebo \
+        #                      and delete the entity when this program is interrupted')
+        self.args = parser.parse_args(args[1:])
+
+        # TODO(shivesh): Wait for /set_model_configuration
+        # (https://github.com/ros-simulation/gazebo_ros_pkgs/issues/779)
+        # Convert position of joints to floats
+        # for i in range(len(self.args.joints)):
+        #     self.args.joints[i][1] = float(self.args.joints[i][1])
+
+    def run(self):
+        """
+        Run node, spawning entity and doing other actions as configured in program arguments.
+
+        Returns exit code, 1 for failure, 0 for success
+        """
+        # Wait for entity to exist if wait flag is enabled
+        if self.args.wait:
+            self.entity_exists = False
+
+            def entity_cb(entity):
+                self.entity_exists = self.args.wait in entity.name
+
+            self.subscription = self.create_subscription(
+                ModelStates, '%s/model_states' % self.args.gazebo_namespace, entity_cb, 10)
+
+            self.get_logger().info(
+                'Waiting for entity {} before proceeding.'.format(self.args.wait))
+
+            while rclpy.ok() and not self.entity_exists:
+                rclpy.spin_once(self)
+                pass
+
+        # Load entity XML from file
+        if self.args.file:
+            self.get_logger().info('Loading entity XML from file %s' % self.args.file)
+            if not os.path.exists(self.args.file):
+                self.get_logger().error('Error: specified file %s does not exist', self.args.file)
+                return 1
+            if not os.path.isfile(self.args.file):
+                self.get_logger().error('Error: specified file %s is not a file', self.args.file)
+                return 1
+            # load file
+            try:
+                f = open(self.args.file, 'r')
+                entity_xml = f.read()
+            except IOError as e:
+                self.get_logger().error('Error reading file {}: {}'.format(self.args.file, e))
+                return 1
+            if entity_xml == '':
+                self.get_logger().error('Error: file %s is empty', self.args.file)
+                return 1
+        # Load entity XML from file published on topic specified in ROS param
+        elif self.args.param:
+            self.get_logger().info(
+                'Loading entity XML from file published on topic %s' % self.args.param)
+            entity_xml = ''
+
+            def entity_xml_cb(entity_xml_path):
+                nonlocal entity_xml
+                try:
+                    f = open(entity_xml_path.data, 'r')
+                    entity_xml = f.read()
+                except IOError as e:
+                    self.get_logger().error(
+                        'Error reading file {}: {}'.format(entity_xml_path.data, e))
```

Comment:
Sorry, my bad, I gave you bad guidance on my previous comment. :bowing_woman: 

I read https://github.com/ros2/rviz/pull/378 too quickly and thought that the standard now is to send the file path, but actually that PR reverted that behaviour to work like ROS 1. So in summary, we should treat the incoming data as the xml, not the path.